### DISCUSSION TITLE
Nextjs 13 "App" directory. Any plans to support supabase-auth-nextjs with generateStaticParams and caching

### DISCUSSION THREAD

**Discussion by Metarock** [supabase/supabase#14378] [Questions]
## Information 
`@supabase/auth-helpers-nextjs`: ^0.6.0
`@supabase/supabase-js`: ^2.19.0
`next`: 13.4.2

## Question/issue 
I want to use Nextjs 13's `generateStaticParams` to build the app and utilise its caching API. https://nextjs.org/docs/app/api-reference/functions/generate-static-params. 

However, because the way I have supabase auth set up for Nextjs based on this [guide](https://supabase.com/docs/guides/auth/auth-helpers/nextjs-server-components). The app would disable caching per request as it is using `next/headers` to get the `cookies` and `headers` in the server component, as stated in this [doc](https://nextjs.org/docs/app/building-your-application/data-fetching/caching). I'm just wondering if there are any plans to provide alternate implementation for supabase nextjs? 

```
export default async function RootLayout({ children }) {
  const supabase = createServerComponentSupabaseClient<Database>({
    headers,
    cookies,
  })
  const {
    data: { session },
  } = await supabase.auth.getSession()
  return (
    <html lang="en">
      <body
        className="min-h-screen bg-white text-slate-900"
      >
        <SupabaseProvider session={session}>
          <main>{children}</main>
        </SupabaseProvider>
      </body>
    </html>
  )
}
```

**Comment by jem-computer** [comment]
+++ this - I just started using Supabase today(!! it's so sick!) and passing cookies into createServerComponentSupabaseClient causes this, which kinda defeats the purpose of server components!

```
[    ] - info Generating static pages (0/5)- warn Entire page / deopted into client-side rendering.
https://nextjs.org/docs/messages/deopted-into-client-rendering /
```

looks like it's related to https://github.com/vercel/next.js/issues/48442

**Comment by dmitkov28** [comment]
Hey, any updates/workarounds on this?
